pkg/apis/discovery/validation/validation.go

```diff
@@ -110,6 +120,17 @@ func validateEndpoints(endpoints []discovery.Endpoint, addrType discovery.Addres
 			}
 		}
 
+		if endpoint.NodeName != nil {
+			nnPath := idxPath.Child("nodeName")
+			if allowNodeName {
+				for _, msg := range apivalidation.ValidateNodeName(*endpoint.NodeName, false) {
+					allErrs = append(allErrs, field.Invalid(nnPath, *endpoint.NodeName, msg))
+				}
+			} else {
+				allErrs = append(allErrs, field.Forbidden(nnPath, "may not be set unless EndpointSliceNodeName feature gate is enabled"))
```

Comment:
I didn't see anything in EndpointSlice strategy that would drop this field when the gate is enabled. This means we would reject clients on validation instead of dropping the disable field. Is that expected?

Reply:
Yep, that's correct, this relies on validation and will fail instead of dropping the field. This matches the [approach taken for AppProtocol](https://github.com/kubernetes/kubernetes/blob/release-1.18/pkg/apis/core/validation/validation.go#L4180-L4187) and some of the other new fields I've seen. I personally think it's more clear to fail upfront than to silently drop fields. 

I'm not sure if there's a standard written anywhere though, I noticed you're dropping the new `terminating` and `serving` conditions in strategy and have not added anything to prevent them in validation. Either way, it seems like we should probably be using the same approach for both cases. @liggitt or @thockin which approach should we be using here for new fields (dropping in strategy or blocking in validation)?

Reply:
Nevermind, @aojea provided a good [reference link](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#alpha-field-in-existing-api-version) above for why this should be added to strategy, working on that now.